File: Front-End/src/services/feedback.service.ts
import axiosClient from '@/configurations/axios.config';
import type {
  FeedbackResponse,
  FeedbackListResponse,
  FeedbackFilters
} from '@/types/feedback.type';

export const feedbackService = {
  getAllFeedbacks: async (page: number = 1, size: number = 10, filters?: FeedbackFilters) => {
    let url = `/feedbacks?page=${page}&size=${size}`;

    if (filters?.rating !== undefined) {
      url += `&rating=${filters.rating}`;
    }
    if (filters?.status !== undefined && filters?.status !== null) {
      url += `&status=${filters.status}`;
    }
    if (filters?.fromDate) {
      url += `&fromDate=${filters.fromDate}`;
    }
    if (filters?.toDate) {
      url += `&toDate=${filters.toDate}`;
    }

    const response = await axiosClient.get<FeedbackListResponse>(url);
    return response.data;
  }, getFeedbackById: async (id: number) => {
    const response = await axiosClient.get<FeedbackResponse>(`/feedbacks/${id}`);
    return response.data;
  },

  changeStatusFeedback: async (id: number) => {
    await axiosClient.put(`/feedbacks/change-status/${id}`);
  },

  deleteFeedback: async (id: number) => {
    await axiosClient.delete(`/feedbacks/${id}`);
  },
};

File: Back-End/src/main/java/iuh/fit/ecommerce/services/FeedbackService.java
package iuh.fit.ecommerce.services;

import iuh.fit.ecommerce.dtos.request.feedback.CreateFeedbackRequest;
import iuh.fit.ecommerce.dtos.response.feedback.FeedbackResponse;
import iuh.fit.ecommerce.dtos.response.feedback.RatingStatisticsResponse;
import org.springframework.data.domain.Page;

public interface FeedbackService {
    FeedbackResponse createFeedback(CreateFeedbackRequest request);
    boolean checkIfReviewed(Long orderId, Long productVariantId);
    FeedbackResponse getFeedbackDetail(Long orderId, Long productVariantId);
    
    // Customer/Public APIs
    Page<FeedbackResponse> getFeedbacksByProduct(Long productId, int page, int size);
    RatingStatisticsResponse getRatingStatistics(Long productId);
    
    // Admin APIs
    Page<FeedbackResponse> getAllFeedbacks(int page, int size, Integer rating, Boolean status, String fromDate, String toDate);
    FeedbackResponse getFeedbackById(Long id);
    void changeStatusFeedback(Long id);
    void deleteFeedback(Long id);
}
